odoo/tools/mail.py

```diff
@@ -304,7 +304,7 @@ def html2plaintext(html, body_id=None, encoding='utf-8'):
 
     html = ustr(html)
 
-    if not html:
+    if not html or html.isspace():
```

Comment:
Could be great to move it in its own commit + add a test (in odoo/addons/base/tests/test_mail.py I think, we already test plaintext2html for example in that file).

Reply:
Actually I'm sorry this condition is not useful anymore. I must have simply forgot to remove it after resolving the problem another way (with api.constrains). The SMS will now never get to this point with white spaces so the check is unnecessary.